data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -266,12 +255,13 @@ public void testRandomData() throws IOException {
     append.commit();
 
     Set<Record> records = Sets.newHashSet(IcebergGenerics.read(table).build());
-    Assert.assertEquals(
-        "Should produce correct number of records", expected.size(), records.size());
-    Assert.assertEquals("Random record set should match", Sets.newHashSet(expected), records);
+    Assertions.assertThat(expected).as("Should produce correct number of records")
```

Comment:
should be `assertThat(records).hasSameSizeAs(expected)`. Please also update this in other places